src/tests/system/tests/test_default_debug_level.py

```diff
@@ -0,0 +1,196 @@
+
+        starting = re.compile(r"Starting with debug level = 0x0070")
+        assert starting.search(log_str) is not None, f"Log file has wrong format: {log_str}"
+
+        log_lines = log_str.split("\n")
+        if len(log_lines) <= 1:
+            continue
+
+        for line in log_lines:
+            searched = re.compile(r"\(0x\w+\)").search(line)
+            if searched is not None and searched.group() == "(0x3f7c0)":
+                check = True
+                break
+        if not check:
+            assert False, "Wrong format of log file"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_default_debug_level__check_with_login(client: Client, provider: GenericProvider):
+    """
+    :title: Successful login with default debug level doesn't generate any logs
+    :setup:
+        1. Add user to SSSD, set its password
+        2. Add fallback_homedir (generates extra logs on
+            user auth if not specified)
+    :steps:
+        1. Clear cache and logs
+        2. Start SSSD with default debug level
+        3. Check total log size before user authentication
+        4. Authenticate with user
+        5. Check total log size after authentication
+    :expectedresults:
+        1. Logs cleared
+        2. SSSD started successfully
+        3. Checked correctly
+        4. User is authenticated
+        5. Log sizes before and after are same
+    :customerscenario: False
+    """
+
+    def log_length() -> int:
+        try:
+            output = client.fs.wc("/var/log/sssd/*", lines=True).stdout
+            return int(output.split()[0])
+        except SSHProcessError:
+            return 0
+
+    provider.user("user1").add(password="Secret123")
+    client.sssd.common.local()
+    client.sssd.default_domain = "local"
+    client.sssd.domain["fallback_homedir"] = "/home/%%u"
+
+    client.sssd.clear(db=True, memcache=True, logs=True, config=False)
+    client.sssd.start(debug_level=None)
+
+    start_log_length = log_length()
+    assert client.auth.ssh.password("user1", "Secret123"), "Authentication failed"
+    assert start_log_length == log_length(), "Debug messages were generated"
+
+
+@pytest.mark.ticket(bz=1893159)
+@pytest.mark.topology(KnownTopology.Client)
+def test_default_debug_level__fatal_and_critical_failures(client: Client):
+    """
+    :title: Check that messages with levels 0 and 1 are logged
+    :setup:
+        1. Start SSSD with default debug level (config file is created)
+    :steps:
+        1. Restrict sssd.conf permissions
+        2. Restart sssd
+        3. Check logs
+    :expectedresults:
+        1. Permissions changed successfully
+        2. SSSD fails to start
+        3. Fatal failures (level 0) and Critical failures (level 1) are in log file
+    :customerscenario: True
+    """
+    client.sssd.common.local()
+    client.sssd.default_domain = "local"
+    client.sssd.start(debug_level=None)
+
+    assert client.host.ssh.run("chmod 444 /etc/sssd/sssd.conf").rc == 0
```

Comment:
I think this should be moved to some utility class in the testing framework. If not, I suspect the permission ownership won't be reversed once the test is finished.

Reply:
The sssd.conf file might be reverted to original state automatically including ownership:
https://github.com/SSSD/sssd-test-framework/blob/master/sssd_test_framework/utils/sssd.py#L372

Reply:
What exactly is this asserting? You are just changing permissions here, this should be in :setup.

Reply:
I assert that the chmod call was successful. I think it is worth checking. But there should be proper fail message.